crates/runtime/src/dataconnector/spark.rs

```diff
@@ -88,8 +90,24 @@ impl DataConnectorFactory for Spark {
         params: Arc<Option<HashMap<String, String>>>,
     ) -> Pin<Box<dyn Future<Output = super::NewDataConnectorResult> + Send>> {
         Box::pin(async move {
-            let spark_connector = Spark::new(Arc::new(secret), params).await?;
-            Ok(Arc::new(spark_connector) as Arc<dyn DataConnector>)
+            match Spark::new(Arc::new(secret), params).await {
+                Ok(spark_connector) => Ok(Arc::new(spark_connector) as Arc<dyn DataConnector>),
+                Err(e) => match e {
+                    Error::UnableToConstructSparkConnect { source } => {
+                        Err(DataConnectorError::UnableToConnectInternal {
```

Comment:
is this supposed to be configure error, and the below one to be internal?

Reply:
This is supposed to be an internal error. Because before where UnableToConstructSparkConnect emitted all configurations are validated, and those errors relevant to invalid configurations are covered in _ =>  case

Reply:
I get what you mean. let me update something